File: TRPO/lab6/myproject/bin/main/com/example/model/UserType.kt
package com.example.model 

import java.io.InputStream

interface TypeComparator {
    fun compare(o1: Any?, o2: Any?): Int
}

interface UserType {
    fun typeName(): String
    fun create(): Any
    fun clone(obj: Any): Any
    fun readValue(input: InputStream): Any
    fun parseValue(ss: String): Any
    fun getComparator(): TypeComparator
}

File: TRPO/lab5/test/lab5/src/main/scala/UserType.scala
import java.io.{InputStreamReader, Serializable}

trait Comparator extends Serializable {
  def compare(o1: Any, o2: Any): Int
}

trait UserType extends Serializable {
  def typeName: String
  def create: Any
  def clone(obj: Any): Any
  def readValue(in: InputStreamReader): Any
  def parseValue(ss: String): Any
  def getTypeComparator: Comparator
}